docs/designers-developers/developers/tutorials/javascript/extending-the-block-editor.md

```diff
@@ -0,0 +1,41 @@
+
+## Extending the Block Editor
+
+This puts all the initial pieces in place for you to start extending the Block Editor.
+
+Let's look at using the [Block Style Variation example](../../../../../docs/designers-developers/developers/filters/block-filters/#block-style-variations).
+
+Replace the existing `console.log()` code in your `myguten.js` file with:
+
+```js
+wp.blocks.registerBlockStyle( 'core/quote', {
```

Comment:
After this piece of code, we could introduce two things: 

- that WordPress makes Gutenberg packages available in the browser through the `wp.` namespace.
- that our plugin needs to tell WordPress what packages we want to use (add dependencies to the enqueueing mechanism).

GitHub still don't let me suggest anything longer than oneliners, but here's a suggestion:

---

Replace the existing `console.log()` code in your `myguten.js` file with:

```js
wp.blocks.registerBlockStyle( 'core/quote', {
    name: 'fancy-quote',
    label: 'Fancy Quote',
} );
```

Update the code in `guten-plugin.php` to be:

```php
/**
 * Plugin Name: My Guten Plugin
 */
function myguten_enqueue() {
	wp_enqueue_script( 'myguten-script',
		plugins_url( 'myguten.js', __FILE__ ),
		array( 'wp-blocks' )
	);
}
add_action( 'enqueue_block_editor_assets', 'myguten_enqueue' );
```

In the [`wp_enqueue_script`](https://developer.wordpress.org/reference/functions/wp_enqueue_script/) function, we declare that our plugin depends on the wp-blocks package (see [Packages](https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/)). WordPress will make that available for us using the `wp` namespace, so, in the browser we can use `wp.blocks` to access anything that package exports.

After these changes, save the files, and then create a new post in the Block Editor.

Reply:
Thanks, this is great, I reworded it some and updated to fit with the flow.